waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -530,8 +556,22 @@ proc relayConnectivityLoop*(pm: PeerManager) {.async.} =
     await pm.connectToRelayPeers()
     await sleepAsync(ConnectivityLoopInterval)
 
+proc updateMetrics(pm: PeerManager) {.async.} =
+  # TODO: Hardcoded due to circular dependency
+  let WakuFilterCodec = "/vac/waku/filter/2.0.0-beta1"
+  let WakuLightPushCodec = "/vac/waku/lightpush/2.0.0-beta1"
+
+  heartbeat "Scheduling updateMetrics run", UpdateMetricsInterval:
+    for dir in @[Direction.In, Direction.Out]:
+      for proto in @[WakuRelayCodec, WakuStoreCodec, WakuFilterCodec, WakuLightPushCodec]:
```

Comment:
To me it's a bit weird that the peer manager would have knowledge of these protocols. Why couldn't it just report against the unique set of protocols retrievable from the `Protobook`? This way this list can dynamically change. If need be, we may even considering filtering only for protocols prefixed `/vac/waku`, so that we're sure we don't report any unrelated libp2p proto streams.

Reply:
Another option, which may be a bit complex for this PR, could be to provide a method by which protocols can register themselves on the peer manager. That way they could not only register their codec for reporting, but perhaps also a retention policy for peers in the store (e.g. max number of filter peers to store).

Reply:
good point. fixed in c3d7a3b9584a7d7a64346966fb0e886a8980f18a

now the list of protocols can dynamically change and the peer manager doesnt need knowledge of them

thanks!